credentialproviderpackage/cmd/aws-credential-provider/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	_ "embed"
+	"github.com/fsnotify/fsnotify"
+	"io/fs"
+	"os"
+	"strings"
+
+	cfg "credential-provider/pkg/configurator"
+	"credential-provider/pkg/configurator/bottlerocket"
+	"credential-provider/pkg/configurator/linux"
+	"credential-provider/pkg/constants"
+	"credential-provider/pkg/log"
+)
+
+func main() {
+	var configurator cfg.Configurator
+	osType := strings.ToLower(os.Getenv("OS_TYPE"))
+	if osType == "" {
+		log.ErrorLogger.Println("Missing Environment Variable OS_TYPE")
+		os.Exit(1)
+	}
```

Comment:
This is fine, why not command line args?

Reply:
I find command line args are more difficult to test with tools such as scaffold where they try to override the entrypoint with a different one for debugging. 

If there is strong feelings towards this let me know and I can change it to command line